influxdb3/tests/server/auth.rs

```diff
@@ -144,13 +99,101 @@ async fn auth() {
     );
     assert_eq!(
         client
-            .get("http://127.0.0.1:8181/api/v3/query_sql?db=foo&q=select+*+from+cpu")
+            .get(format!("{base}/api/v3/query_sql"))
+            .query(&[("db", "foo"), ("q", "select * from cpu")])
             .header("auth", format!("Bearer {TOKEN}"))
             .send()
             .await
             .unwrap()
             .status(),
         StatusCode::UNAUTHORIZED
     );
-    COMMAND.lock().take().unwrap().kill();
+}
+
+#[tokio::test]
+async fn auth_grpc() {
+    const HASHED_TOKEN: &str = "5315f0c4714537843face80cca8c18e27ce88e31e9be7a5232dc4dc8444f27c0227a9bd64831d3ab58f652bd0262dd8558dd08870ac9e5c650972ce9e4259439";
+    const TOKEN: &str = "apiv3_mp75KQAhbqv0GeQXk8MPuZ3ztaLEaR5JzS8iifk1FwuroSVyXXyrJK1c4gEr1kHkmbgzDV-j3MvQpaIMVJBAiA";
+
+    let server = TestServer::configure()
+        .auth_token(HASHED_TOKEN, TOKEN)
+        .spawn()
+        .await;
+
+    // Write some data to the server, this will be authorized through the HTTP API
+    server
+        .write_lp_to_db(
+            "foo",
+            "cpu,host=s1,region=us-east usage=0.9 1\n\
+            cpu,host=s1,region=us-east usage=0.89 2\n\
+            cpu,host=s1,region=us-east usage=0.85 3",
+            Precision::Nanosecond,
+        )
+        .await
+        .unwrap();
+
+    {
+        // Spin up a FlightSQL client
+        let mut client = server.flight_sql_client("foo").await;
+
+        // Make a query using the client, without providing any authorization header:
+        let error = client.query("SELECT * FROM cpu").await.unwrap_err();
+        assert!(matches!(error, FlightError::Tonic(s) if s.code() == tonic::Code::Unauthenticated));
+
+        // Set the authorization header on the client:
+        client
+            .add_header("authorization", &format!("Bearer {TOKEN}"))
```

Comment:
I think hyper works either way here, but I think we should capitalize it to `Authorization` for consistency's sake.

Reply:
In `service_grpc_flight`, `"authorization"` is used, but I agree - I think it is supposed to be case insensitive, so I will add tests for both `A` and `a` and see what happens.